### PR TITLE
Setup, Test and Travis CI setup

### PR DESCRIPTION
**Pull request by vbsinha**
This work is related to setting up this repo and properly adding tests. The changes to be made include:

- [x] Create the directory structure similar to pymc3 and place the files and tests appropriately.
- [x] Add setup.py and corresponding files like requirements.txt etc. 
- [x]  Setup Travis CI and the corresponding scripts in scripts directory.
- [x] Modify the notebooks to use SGFS and CSG from `pymc3_experimental`.

Package names can not contain '-' in python so I have named the package `pymc3_experimental` right now. @twiecki Is this ok?

I am working on Travis CI test scripts. Currently the test case can be run using `pytest -v`.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+import re
+
+DISTNAME = 'pymc3_experimental'
+DESCRIPTION = "Experimental Modules of pymc3 (Probabilistic Programming in Python: Bayesian Modeling and Probabilistic Machine Learning with Theano)"
+AUTHOR = 'PyMC Developers'
+AUTHOR_EMAIL = 'pymc.devs@gmail.com'
+URL = "http://github.com/pymc-devs/pymc3-experimental"
+LICENSE = "Apache License, Version 2.0"
+
+classifiers = ['Development Status :: 5 - Production/Stable',
```
Comment (twiecki) [inline setup.py]: should probably not have this label :)

**Comment by twiecki** [comment]
This looks great so far!

File: setup.py (hunk 1 above)
Comment (vbsinha) [reply setup.py]: Thanks for pointing out. I have changed it to `Development Status :: 2 - Pre-Alpha`. Is that ok or shall I remove this line altogether?

**Comment by vbsinha** [comment]
I have modified scripts for Travis CI for this repo (Mostly copied from `pymc3` repo). Hopefully, they should work!! Could you try integrating Travis CI with this repo?

**Comment by twiecki** [comment]
Looks great, I activated travis. Not sure if it will show up for this PR though.

**Comment by ColCarroll** [comment]
Cool! Can have a followup adding the badge to the readme:

https://travis-ci.org/pymc-devs/pymc3-experimental

**Comment by ColCarroll** [comment]
they all failed because coveralls isn't active. i activated coveralls and restarted travis.

**Pull request merged**